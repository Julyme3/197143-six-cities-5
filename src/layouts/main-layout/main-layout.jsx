import React, {memo} from "react";
import PropTypes from "prop-types";
import Header from "../../components/header/header";

const MainLayout = (props) => {
  const {children, className = ``, classNameWrap = ``} = props;

  return (
    <div className={`page ${className}`}>
      <Header />
      <main className={`page__main ${classNameWrap}`}>
        {children}
      </main>
    </div>
  );
};

MainLayout.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  classNameWrap: PropTypes.string,
};

export {MainLayout};
export default memo(MainLayout);
