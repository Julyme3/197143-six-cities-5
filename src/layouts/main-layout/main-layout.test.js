import React from "react";
import renderer from "react-test-renderer";
import {MainLayout} from "./main-layout";

jest.mock(`../../components/modal-notfication/modal-notification`, () => `ModalNotification`);
jest.mock(`../../components/header/header`, () => `Header`);

test(`MainLayout render`, () => {
  const tree = renderer
    .create(
        <MainLayout
          className="className"
          classNameWrap="classNameWrap"
        >
          <React.Fragment />
        </MainLayout>
    )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
