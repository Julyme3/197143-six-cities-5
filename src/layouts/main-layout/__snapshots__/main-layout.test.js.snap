// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout render 1`] = `
<div
  className="page className"
>
  <Header />
  <main
    className="page__main classNameWrap"
  />
  <ModalNotification />
</div>
`;
